src/replication.c

```diff
@@ -845,22 +846,24 @@ static void appendFollowerCb(struct raft_io_append *req, int status)
     assert(args->entries != NULL);
     assert(args->n_entries > 0);
 
-    result.term = r->current_term;
-    result.version = RAFT_APPEND_ENTRIES_RESULT_VERSION;
-    if (status != 0) {
-        if (r->state != RAFT_FOLLOWER) {
-            tracef("local server is not follower -> ignore I/O failure");
-            goto out;
-        }
-        result.rejected = args->prev_log_index + 1;
-        goto respond;
-    }
-
     /* If we're shutting down or have errored, ignore the result. */
     if (r->state == RAFT_UNAVAILABLE) {
         tracef("local server is unavailable -> ignore I/O result");
         goto out;
     }
+    /* If we've transitioned to candidate state since the request came in,
+     * ignore the result. */
+    if (request->term != r->current_term) {
+        tracef("state transition while processing request -> ignore I/O result");
+        goto out;
+    }
+
+    result.version = RAFT_APPEND_ENTRIES_RESULT_VERSION;
+    result.term = request->term;
+    if (status != 0) {
+        result.rejected = args->prev_log_index + 1;
+        goto respond;
```

Comment:
I think we can respond here after converting to `RAFT_UNAVAILABLE` which is unwanted.

Reply:
yeah, sorry, looked over that one.